packages/xml-views-validation/src/validators/attributes/none-unique-id.ts

```diff
@@ -0,0 +1,79 @@
+import { clone, flatMap, map, omitBy } from "lodash";
+import {
+  accept,
+  XMLAstVisitor,
+  XMLAttribute,
+  XMLDocument,
+  XMLToken,
+} from "@xml-tools/ast";
+import { UI5SemanticModel } from "@ui5-language-assistant/semantic-model-types";
+import { NoneUniqueIDIssue } from "../../../api";
+
+export function validateUniqueID(
+  xmlDoc: XMLDocument,
+  model: UI5SemanticModel
+): NoneUniqueIDIssue[] {
+  const idCollector = new IdsCollectorVisitor();
+  accept(xmlDoc, idCollector);
+  const idsToXMLElements = idCollector.idsToXMLElements;
+
+  const duplicatedIdsRecords = omitBy(idsToXMLElements, (_) => _.length > 1);
+  const issues: NoneUniqueIDIssue[] = flatMap(
+    duplicatedIdsRecords,
+    (currDupAttribsPerID, id) => {
+      return map(currDupAttribsPerID, (currDupAttrib, currIdx) => {
```

Comment:
This is a bit complex. Could you add a comment that it maps each duplicate ID attribute to an issue, then flattens the list of issues-list-per-id? Or maybe add a named function `createIssueForDuplicateID(id, allDuplicateIDAttributes, currAttribute, currAttributeIndex)` that returns the issue and call it from here?

Reply:
refactored to simplify things (new named function to handle a single ID duplications)